examples/experimental/script_to_assets_branching_io_manager/hn_dagster_external_execution.py

```diff
@@ -0,0 +1,140 @@
+import json
+import subprocess
+import sys
+from typing import List, Optional, Sequence, Set
+
+from dagster import (
+    AssetsDefinition,
+    Definitions,
+    Nothing,
+    OpExecutionContext,
+    asset,
+    define_asset_job,
+)
+from dagster._core.definitions.events import AssetKey
+from dagster._core.instance import DagsterInstance
+from dagster._core.storage.branching.branching_io_manager import (
+    get_text_metadata_value,
+    latest_materialization_log_entry,
+)
+from dagster._utils import file_relative_path
+
+STORAGE_NAMESPACE_METADATA_KEY = "storage_namespace"
+
+
+def upstream_asset_partition_materialized_in_storage_namespace(
+    instance: DagsterInstance,
+    upstream_asset_key: AssetKey,
+    partition_key: Optional[str],
+    storage_namespace: str,
+):
+    event_log_entry = latest_materialization_log_entry(
+        instance=instance,
+        asset_key=upstream_asset_key,
+        partition_key=partition_key,
+    )
+    return (
+        event_log_entry
+        and event_log_entry.asset_materialization
+        and get_text_metadata_value(
+            event_log_entry.asset_materialization, STORAGE_NAMESPACE_METADATA_KEY
+        )
+        == storage_namespace
+    )
+
+
+def in_branch():
+    return True
```

Comment:
In a real scenario, this would check for the dagster cloud branch deployment env var